doc/design/web.md

```diff
@@ -0,0 +1,141 @@
+# Web Interface design
+
+This design doc will talk about features and web pages needed to let users to manage cloud paddle jobs.
+
+## Feature List
+
+- Account Management
+    - Registration, send email to inform if registration suceeded
+    - Account Login/Logout
+    - Password changing, find back
+    - Download SSL keys
+- Jupiter Notebook
+    - Private Jupiter Notebook environment to run Python scripts
+    - Private workspace
+    - Submit job from Jupiter Notebook
+- Job Dashboard
+    - Job history and current running jobs
+    - Performance Monitoring
+    - Quota Monitoring
+- Datasets
+    - Public Dataset viewing
+    - Upload/Download private datasets
+    - Share datasets
+- Models
+    - Upload/Download models file
+    - Share/Publish Models
+- Paddle Board
+    - Training metrics visualization
+        - cost
+        - evaluator
+        - user-defined metrics
+- Serving
+    - Submit serving instances
+    - Deactivate serving
+    - Serving performance monitoring
+
+## Account Management
+
+I'll skip this section because it is a design that almost every website need.
+
+## Jupiter Notebook
+
+Start a ReplicaSet using image `docker.paddlepaddle.org/book` in kubernetes cluster and add an ingress endpoint when user first enters the notebook page.
+
+<img src="pictures/notebook.png" width="500px" align="center">
+
+Users can write program in python in the web page and save their programs, which will be saved at cloud storage. Users also can run script like below to submit a cluster training job:
+
+```python
+sess = paddle.framework.remote_session(
+    topology=block,
```

Comment:
What does this example program mean? Is it intended to run a block? I am not sure if our API design could generate a block which is assignable to the `topology` parameter.  Basically, our API is designed to generate a `ProgramDesc` protobuf message that includes a repeated field of `BlockDesc` messages, as described here https://github.com/PaddlePaddle/Paddle/blob/develop/doc/design/program.md.

Reply:
Thanks, will just use pseudo code from https://github.com/PaddlePaddle/Paddle/blob/develop/doc/design/refactor/session.md